javascript/javascript1/week3/song.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+const songDatabase = [{
+    songId: 1,
+    title: 'My baby',
+    artist: 'Soggy socks',
+  },
+  {
+    songId: 2,
+    title: '3 nails in wood',
+    artist: 'The carpenters',
+  },
+  {
+    songId: 3,
+    title: 'Blacker than black',
+    artist: 'Instant coffee',
+  },
+  {
+    songId: 4,
+    title: 'When is enough too little?',
+    artist: 'The spies girls',
+  }
+];
+
+
+function addSongToDatabase(song) {
+    const myPlaylist = [];
+    
+    songDatabase.push(song);
+    console.log(songDatabase);
+}
+
+function getSongByTitle(title) {
+  
+       const index = songDatabase.findIndex( function (songDatabase) {
+        return title === songDatabase.title ;
+     })
+
+    return songDatabase[index];
+}
+
+function addSongToMyPlaylist(title) {
```

Comment:
this does not add anything to the playlist. It just returns the song.

Reply:
function addSongToMyPlaylist(title) {
    
  const mySongList = [];
  const titleTrack = getSongByTitle(title);
    mySongList.push(titleTrack);
    return mySongList ;
}

Reply:
Here you're creating the list (from scratch) and adding one song to it. So you are always returning one song. You should probably define mySongList outside the function, and instead of returning, modify it. Perhaps you should brush up on function and variable **scope**.

Reply:
const mySongList = [];

function addSongToMyPlaylist(title) {
    
  
  const titleTrack = getSongByTitle(title);
    mySongList.push(titleTrack);
    
}

Reply:
addSongToMyPlaylist('3 nails in wood');

addSongToMyPlaylist('Happy');
console.log("My PlayList : ",mySongList);

Reply:
Yes that's more like it. And most importantly, does it work?

Reply:
Yes it works.sent screenshots by mail